src/Inventory/Asset/NetworkPort.php

```diff
@@ -870,26 +870,28 @@ public function handleHub($found_macs, $netports_id)
         }
 
         if (!$hubs_id) {
-           //create direct connection
+            //create direct connection if import_unmanaged is enabled
             $hubs_id = $link->createHub($netports_id, $this->entities_id);
         }
 
-        $glpi_ports = [];
-        $dbports = $netport->find([
-            'items_id' => $hubs_id,
-            'itemtype' => Unmanaged::getType()
-        ]);
-        foreach ($dbports as $dbport) {
-            $id = $link->getOppositeContact($dbport['id']);
-            if ($id) {
-                $glpi_ports[$id] = $dbport['id'];
+        if (!$hubs_id) {
```

Comment:
Did you inverse the condition?
```suggestion
        if ($hubs_id) {
```

If so, maybe just exist the method when it's not set rather than including the rest of the code in a global condition.

Reply:
Simply like that it's ok?
```
if (!$hubs_id) {
    //create direct connection if import_unmanaged is enabled
    $hubs_id = $link->createHub($netports_id, $this->entities_id);
    if (!$hubs_id) {
        return;
    }
}
```